components/feature/addons/src/main/java/mozilla/components/feature/addons/migration/SupportedAddonsChecker.kt

```diff
@@ -81,7 +84,7 @@ class DefaultSupportedAddonsChecker(
     override fun registerForChecks() {
         WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
                 CHECKER_UNIQUE_PERIODIC_WORK_NAME,
-                ExistingPeriodicWorkPolicy.REPLACE,
+                ExistingPeriodicWorkPolicy.KEEP,
```

Comment:
This will prevent the worker from running multiple times see, now if we call this function again, it will keep the existing worker in the work manager databes https://developer.android.com/reference/androidx/work/ExistingPeriodicWorkPolicy#KEEP.

[Before we were deleting the exiting work and adding a new one](https://developer.android.com/reference/androidx/work/ExistingPeriodicWorkPolicy#REPLACE), as Fenix  executes `registerForChecks` every time the app starts, the conditions were met and we were executing the worker every time.